src/amo/reducers/currentView.js

```diff
@@ -0,0 +1,38 @@
+/* @flow */
+import { CURRENT_VIEW_SET } from 'core/constants';
+
+
+export type currentViewTypes = {|
+  // TODO: Create an addonType Type that is one of type (ADDON_TYPE_EXTENSION,
+  // ADDON_TYPE_THEME, etc.)
+  addonType: string | null,
+  isExploring: boolean | void,
```

Comment:
Instead of `addonType` and `isExploring`, couldn't you just have a single value that indicates what the context is? I think this would mirror the menu UI a bit more. Example:

```js
type ViewContextState = {|
  context: 'exploring' | 'extensions' | 'themes',
|}
```

You could still support dispatches like `setAddonType(addonType)` and just reduce them as needed, like:

```js
switch (action.type) {
  case 'SET_ADDON_TYPE':
    switch(action.payload.addonType) {
      case 'extension':
        return { ...state, context: 'extensions' };      
      case 'persona':
        return { ...state, context: 'themes' };
    }
}

```